iocore/eventsystem/unit-tests/test_MIOBufferWriter.cc

```diff
@@ -156,62 +157,61 @@ writeOnce(MIOBufferWriter &bw, std::size_t len)
   REQUIRE(bw.auxBufferCapacity() <= BlockSize);
 }
 
-class InkAssertExcept
-{
-};
+std::jmp_buf jmpBuf;
 
 TEST_CASE("MIOBufferWriter", "[MIOBW]")
 {
   MIOBufferWriter bw(theMIOBuffer);
 
-  REQUIRE(bw.auxBufferCapacity() == BlockSize);
-
-  writeOnce(bw, 0);
-  writeOnce(bw, 1);
-  writeOnce(bw, 1);
-  writeOnce(bw, 1);
-  writeOnce(bw, 10);
-  writeOnce(bw, 1000);
-  writeOnce(bw, 1);
-  writeOnce(bw, 0);
-  writeOnce(bw, 1);
-  writeOnce(bw, 2000);
-  writeOnce(bw, 69);
-  writeOnce(bw, 666);
-
-  for (int i = 0; i < 3000; i += 13) {
-    writeOnce(bw, i);
-  }
+  if (setjmp(jmpBuf)) {
+    // Unexpectedly triggered ink_assert().
+    REQUIRE(false);
 
-  writeOnce(bw, 0);
-  writeOnce(bw, 1);
+  } else {
+    REQUIRE(bw.auxBufferCapacity() == BlockSize);
+
+    writeOnce(bw, 0);
+    writeOnce(bw, 1);
+    writeOnce(bw, 1);
+    writeOnce(bw, 1);
+    writeOnce(bw, 10);
+    writeOnce(bw, 1000);
+    writeOnce(bw, 1);
+    writeOnce(bw, 0);
+    writeOnce(bw, 1);
+    writeOnce(bw, 2000);
+    writeOnce(bw, 69);
+    writeOnce(bw, 666);
+
+    for (int i = 0; i < 3000; i += 13) {
+      writeOnce(bw, i);
+    }
 
-  REQUIRE(bw.extent() == ((iobbIdx * BlockSize) + blockUsed));
+    writeOnce(bw, 0);
+    writeOnce(bw, 1);
 
-// These tests don't work properly with clang for some reason.
-#if !defined(__clang__)
+    REQUIRE(bw.extent() == ((iobbIdx * BlockSize) + blockUsed));
+  }
 
-  try {
+  if (!setjmp(jmpBuf)) {
+    // This should cause ink_assert().
     bw.write(bw.auxBufferCapacity() + 1);
```

Comment:
This can be handled by setting a global when call ink_assert() and then returning from the method.

Reply:
But generally in our code ink_assert() is treated as [[noreturn]], so in a mocked version of ink_assert() in UT, I don't see any alternative to an exception throw or longjmp().  Unless I just eliminate the tests that trigger ink_assert().  Which in general will mean we can't get 100% test coverage.